stan/math/opencl/matrix_cl.hpp

```diff
@@ -46,11 +49,50 @@ class matrix_cl {
   int cols() const { return cols_; }
 
   int size() const { return rows_ * cols_; }
-
   const cl::Buffer& buffer() const { return oclBuffer_; }
-
   matrix_cl() : rows_(0), cols_(0) {}
 
+  matrix_cl(cl::Buffer& A, const int& R, const int& C) : rows_(R), cols_(C) {
+    oclBuffer_ = A;
+  }
+  template <typename T>
+  explicit matrix_cl(std::vector<T> A) : rows_(1), cols_(A.size()) {
+    if (A.size() == 0)
+      return;
+    // the context is needed to create the buffer object
+    cl::Context& ctx = opencl_context.context();
+    cl::CommandQueue& queue = opencl_context.queue();
+    try {
+      oclBuffer_
+          = cl::Buffer(ctx, CL_MEM_READ_WRITE, sizeof(double) * A.size());
+      queue.enqueueWriteBuffer(oclBuffer_, CL_TRUE, 0,
+                               sizeof(double) * A.size(),
+                               value_of_rec(A).data());
```

Comment:
when T is double, value_of_rec should just return the reference right? https://github.com/stan-dev/math/blob/develop/stan/math/prim/arr/fun/value_of_rec.hpp#L42